clinica/pipelines/engine.py

```diff
@@ -607,8 +608,31 @@ def check_not_cross_sectional(self):
         from os import listdir
         from os.path import abspath, basename, dirname, isdir, join
 
+        from clinica.utils.exceptions import ClinicaInconsistentDatasetError
         from clinica.utils.stream import cprint
 
+        def detect_cross(all_subs, bids_dir, cross_subj, long_subj):
```

Comment:
I would extract this function out of the `Pipeline` class, add type hints and a proper docstring. 
We can discuss this, but I think this function shouldn't raise/print anything, but return the list of cross sectional and longitudinal subjects.
Maybe the name can be changed to `detect_cross_sectional_and_longitudinal_subjects` or something like that. 
I would also remove the last two input parameters to avoid unnecessary mutations.